dev/com.ibm.ws.app.manager/src/com/ibm/ws/app/manager/internal/lifecycle/ApplicationStartBarrierImpl.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.app.manager.internal.lifecycle;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.wsspi.application.lifecycle.ApplicationPrereq;
+import com.ibm.wsspi.application.lifecycle.ApplicationStartBarrier;
+
+/**
+ * Enable Declarative Services to enforce start sequencing.
+ * Application handlers should depend on this service.
+ * Services should implement {@link ApplicationPrereq}
+ */
+@Component(configurationPolicy = ConfigurationPolicy.REQUIRE)
+public class ApplicationStartBarrierImpl implements ApplicationStartBarrier {
+    private static final TraceComponent tc = Tr.register(ApplicationStartBarrierImpl.class);
+
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE)
+    void setApplicationPrereq(ApplicationPrereq applicationPrereq) {
+    }
+
+    /**
+     * Non configurable immediate component to track prereqs as they appear.
+     */
+    @Component(configurationPolicy = ConfigurationPolicy.IGNORE, immediate = true)
+    public static class PrereqWatcher {
```

Comment:
It is not clear to me how this component does anything other than provide some additional trace points.

Reply:
The test case and https://github.com/OpenLiberty/open-liberty/pull/11152
are (going to be)  applicationPrereqs